### PR TITLE
DSD-1546: UI updates

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes [JIRA ticket DSD-1546](http://jira.nypl.org/browse/DSD-1546)

## This PR does the following:

- Updates the Header based on new designs.
- Noticeable updates:
  - Updated icon for the logged in user
  - smaller height overall
  - new spacing between links in both navs
  - updated `Donate` link button color to be consistent with DS brand primary color
  - new breakpoints specifically for the Header, 832px and 1024px, set as "mh" and lh" for "medium header" and "large header", respectively.

## How has this been tested?

Locally.

### Checklist:

- [ ] I have updated the Readme documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 not completely ready yet but can you do a first pass to check it out?

The new mobile designs for the button nav is not full height (from what I understand in the figma file) so the active state looks off; new on the left and prod on the right:
Screen Shot 2023-08-21 at 2 51 23 PM

File: src/theme/headerLowerNav.ts [hunk 1]
```diff
@@ -14,17 +14,19 @@ const HeaderLowerNav = {
...
-      fontSize: "1",
-      marginBottom: "xxs",
+      marginBottom: "0",
       marginLeft: "auto",
       whiteSpace: "nowrap",
     },
     li: {
-      letterSpacing: ".02em",
-      marginLeft: "xs",
+      marginEnd: { lg: "s", xl: "m" },
```
Comment (bigfishdesign13) [inline src/theme/headerLowerNav.ts]: In the browser, I am seeing this go to `xs` for the breakpoint just above the mobile view.  And that is a bit too small.  Where is that coming from?  I bumped it up to `s` and it still seems to fit.

File: src/theme/headerUpperNav.ts [hunk 2]
```diff
@@ -1,25 +1,25 @@
...
-      textTransform: "uppercase",
+      fontSize: "desktop.body.body2",
+      fontWeight: "medium",
+      marginRight: "s",
+      _last: {
+        marginRight: "0",
+      },
     },
     a: {
       color: "ui.black",
```
Comment (bigfishdesign13) [inline src/theme/headerUpperNav.ts:22]: You many need to add something to override this for the `Donate` link button.

**Review by bigfishdesign13 — COMMENTED** [review]
Looks good.  Still a work in progress, of course.  I left a few comments about spacing and font sizes.

File: src/theme/headerLowerNav.ts [hunk 3]
```diff
@@ -14,17 +14,19 @@ const HeaderLowerNav = {
...
+      _last: {
+        marginRight: "0",
+      },
     },
     "li > a": {
       color: "ui.black",
+      fontSize: "desktop.subtitle.subtitle2",
```
Comment (bigfishdesign13) [inline src/theme/headerLowerNav.ts]: This comment didn't make it into my review.  Maybe I didn't hit the submit button?  Hmm, not sure.

This should be `subtitle1` for the largest breakpoint.

File: src/theme/headerUpperNav.ts (hunk 2 above)
Comment (EdwinGuzman) [reply src/theme/headerUpperNav.ts:22]: The `Donate` link button already renders as white unless if you mean something else?

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 ready for a review

File: src/theme/headerUpperNav.ts (hunk 2 above)
Comment (bigfishdesign13) [reply src/theme/headerUpperNav.ts:22]: I am seeing black button text.

image

File: src/theme/headerLowerNav.ts [hunk 4]
```diff
@@ -14,17 +14,22 @@ const HeaderLowerNav = {
...
-      fontSize: "1",
-      marginBottom: "xxs",
+      marginBottom: "0",
       marginLeft: "auto",
       whiteSpace: "nowrap",
     },
     li: {
-      letterSpacing: ".02em",
-      marginLeft: "xs",
+      marginEnd: { lh: "s", xl: "m" },
```
Comment (bigfishdesign13) [inline src/theme/headerLowerNav.ts]: The code you have here looks correct, but in the browser I am seeing the spacing between the links in the lower nav set to 8px (xs) from the 832px-1023px.

File: src/theme/headerUpperNav.ts (hunk 2 above)
Comment (EdwinGuzman) [reply src/theme/headerUpperNav.ts:22]: Yeah, it's white for me. I wonder if the value somehow got cached _in my browser_? That's not good, but I'll make the color explicit.

Comment (EdwinGuzman) [reply src/theme/headerUpperNav.ts:22]: (I do see it as black in an incognito window.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Overall, looks great.  I left one comment about the spacing between the links in the bottom nav.  

I also think it would be good to adjust the padding on the `My Account` and `Search` links (yes, I know they are actually buttons) to make the focus rectangle the same size as the other links.  What do you think?

**Comment by EdwinGuzman** [comment]
> I also think it would be good to adjust the padding on the My Account and Search links (yes, I know they are actually buttons) to make the focus rectangle the same size as the other links. What do you think?

I like this idea but when you click on them, the background color changes and we'd have to update this as well. It's not the padding in the button but rather the min-height and min-width. Here's an example: 
Screen Shot 2023-08-23 at 11 56 00 AM

If only the height/width is updated on the non-active or non-focus state, then the navs jump around.

**Comment by EdwinGuzman** [comment]
Updated

**Comment by bigfishdesign13** [comment]
@EdwinGuzman 
Some comments came up when I was discussing the `Header` with the designers.
- For desktop, the bottom nav looks like it could be pushed down a bit further so the bottom of the bottom nav is in better alignment with the bottom of the logo.
- The links for `My Account` and `Search` are using Arial font family.
- The sizing of the icons in the mobile menu is not great.  I am going to update some icons and add some new ones to use.

**Comment by EdwinGuzman** [comment]
Did the first two bullet point updates but will need to wait for the new icons.

**Comment by EdwinGuzman** [comment]
Updated with new icons and DS v1.7.2!

**Comment by bigfishdesign13** [comment]
@EdwinGuzman 
The vertical alignment of the `My Account`, `Search` and `hamburger` icons needs to be adjusted in order to better align with the vertical alignment of the `Locations` icon.  Add the following styles to the `<button>` wrappers for those icons.

```
display: flex;
justify-content: center;
```

**Comment by EdwinGuzman** [comment]
Updated.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good!

**Pull request merged**